tensorboard/plugins/interactive_inference/witwidget/notebook/jupyter/js/lib/wit.js

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+var widgets = require('@jupyter-widgets/base');
+
+/**
+ * Helper method to load the vulcanized template.
+ */
+function loadVulcanizedTemplate() {
+    const templateLocation =
+        __webpack_public_path__ + 'wit_jupyter.html';
+
+    // If the vulcanized template is not loaded yet, load it now.
+    if (!document.querySelector('link[href="' + templateLocation + '"]')) {
+        const link = document.createElement('link');
+        link.setAttribute('rel', 'import');
+        link.setAttribute('href', templateLocation);
+        document.head.appendChild(link);
+    }
+}
+
+// What-If Tool View. Renders the tool and provides communication with the
+// python backend.
+var WITView = widgets.DOMWidgetView.extend({
+    render: function() {
+        loadVulcanizedTemplate();
+        this.inferMutantsCounter = 0;
+
+        // Create and attach WIT element to DOM.
+        this.view_ = document.createElement(
+            'tf-interactive-inference-dashboard');
+        this.view_.local = true;
+        this.el.appendChild(this.view_);
+
+        // Add listeners for changes from python.
+        this.model.on('change:examples', this.examples_changed, this);
+        this.model.on('change:config', this.config_changed, this);
+        this.model.on('change:inferences', this.inferences_changed, this);
+        this.model.on('change:eligible_features',
+            this.eligible_features_changed, this);
+        this.model.on('change:mutant_charts', this.mutant_charts_changed, this);
+        this.model.on('change:sprite', this.sprite_changed, this);
+
+        // Add listeners for changes from WIT Polymer element. Passes changes
+        // along to python.
+        this.view_.addEventListener('infer-examples', e => {
```

Comment:
would you be factor out view creation to its own method?

Reply:
done